### PR TITLE
Cache cleanup

### PR DESCRIPTION
**Pull request by davidartplus**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by jijisv** [comment]
if we are changing the plugin name, we should consider changing the package name as well.

File: micro-memcached/src/main/java/com/oath/micro/server/elasticache/ElasticacheConnection.java [hunk 1]
```diff
@@ -4,24 +4,26 @@
 
 import java.util.Optional;
 import net.spy.memcached.MemcachedClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
 
 @Slf4j
-public class TransientElasticacheDataConnection<V> implements DistributedCacheManager<V> {
+public class ElasticacheConnection<K, V> implements DistributedCache<K, V> {
```
Comment (jijisv) [inline micro-memcached/src/main/java/com/oath/micro/server/elasticache/ElasticacheConnection.java:12]: `MemcachedCacheImpl` may be a better name IMO.

File: micro-memcached/src/main/java/com/oath/micro/server/elasticache/ConfigureElasticache.java [hunk 2]
```diff
@@ -39,24 +39,24 @@ public ConfigureElasticache( @Value("${elasticache.hostname:null}") String hostn
 
 
     @Bean(name = "transientCache")
-    public DistributedCacheManager transientCache() throws IOException, URISyntaxException {
+    public DistributedCache transientCache() {
         try {
             log.info("Creating Memcached Data connection for elasticache cluster: {}", hostname);
-            return new TransientElasticacheDataConnection(createMemcachedClient(), retryAfterSecs, maxRetries);
+            return new ElasticacheConnection(createMemcachedClient(), retryAfterSecs, maxRetries);
```
Comment (jijisv) [inline micro-memcached/src/main/java/com/oath/micro/server/elasticache/ConfigureElasticache.java:45]: `DistributedCache` and its impl support generics and we are loosing the type info here..

**Review by jijisv — APPROVED** [review]

File: micro-memcached/src/main/java/com/oath/micro/server/memcached/ElasticachePlugin.java [hunk 3]
```diff
@@ -9,6 +9,6 @@
 
     @Override
     public Set<Class> springClasses() {
-            return SetX.of(ConfigureElasticache.class);
+            return SetX.of(ElasticacheConfig.class);
```
Comment (jijisv) [inline micro-memcached/src/main/java/com/oath/micro/server/memcached/ElasticachePlugin.java:12]: I guess we will have to add `DistributedCacheFactory.class` also here..

File: micro-memcached/src/main/java/com/oath/micro/server/memcached/DistributedCacheFactory.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Autowired
+    public DistributedCacheFactory(ElasticacheConfig config) {
+        this.config = config;
+    }
+
+    public <K, V> DistributedCache<K, V> create() {
+        try {
+            log.info("Creating Memcached Data connection for elasticache cluster: {}", config.getHostname());
+            return new MemcachedCacheImpl(createMemcachedClient(), config.getRetryAfterSecs(), config.getMaxRetries());
```
Comment (jijisv) [inline micro-memcached/src/main/java/com/oath/micro/server/memcached/DistributedCacheFactory.java:25]: I guess
```java
return new MemcachedCacheImpl<K, V>(createMemcachedClient(), ...
```

Comment (davidartplus) [reply micro-memcached/src/main/java/com/oath/micro/server/memcached/DistributedCacheFactory.java:25]: IIRC the compiler can deduce the types <K, V> from the return type.

File: micro-memcached/src/main/java/com/oath/micro/server/memcached/ElasticachePlugin.java (hunk 3 above)
Comment (davidartplus) [reply micro-memcached/src/main/java/com/oath/micro/server/memcached/ElasticachePlugin.java:12]: updated

File: micro-memcached/src/main/java/com/oath/micro/server/memcached/DistributedCacheFactory.java (hunk 4 above)
Comment (davidartplus) [reply micro-memcached/src/main/java/com/oath/micro/server/memcached/DistributedCacheFactory.java:25]: besides, the IDE suggests it's not relevant

Comment (jijisv) [reply micro-memcached/src/main/java/com/oath/micro/server/memcached/DistributedCacheFactory.java:25]: cool, thanks.

**Review by jijisv — APPROVED** [review]

**Review by johnmcclean — APPROVED** [review]
LGTM

**Pull request merged**